src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/PropertyBasedEmbeddingProviderConfig.java

```diff
@@ -29,8 +29,46 @@ interface EmbeddingProviderConfig {
     @JsonProperty
     String apiKey();
 
+    /**
+     * A map of supported authentications. HEADER, SHARED_SECRET and NONE are the only techniques
+     * the DataAPI supports (i.e. the key of map can only be HEADER, SHARED_SECRET or NONE).
+     *
+     * @return
+     */
     @JsonProperty
-    List<String> supportedAuthentication();
+    Map<String, AuthenticationConfig> supportedAuthentication();
+
+    /**
+     * enabled() is a JSON boolean to flag if this technique is supported. If false the rest of the
+     * object has no impact. Any technique not listed is also not supported for the provider.
+     *
+     * <p>tokens() is a list of token mappings, that map from the name accepted by the Data API to
+     * how they are forwarded to the provider. The provider information is included for the code,
+     * and to allow users to see what we do with the values.
+     */
+    interface AuthenticationConfig {
+      @JsonProperty
+      boolean enabled();
+
+      @JsonProperty
+      List<TokenConfig> tokens();
+    }
+
+    /**
+     * For the HEADER technique the `accepted` value is the name of the header the client should
```

Comment:
Why don't we use something like "headerName" instead of "accepted" if this is related to HEADER technique?
And similarly "secretName" instead of "forwarded" when SHARED_SECRET option?

Reply:
Ok so basically this is because both "accepted" and "forwarded" are accepted for both HEADER and SHARED_SECRET. Make sense.